pkg/deploy/deploy.go

```diff
@@ -87,11 +87,57 @@ func (d *deployer) PreDeploy(ctx context.Context) (rpServicePrincipalID string,
 		return "", err
 	}
 
-	deploymentName := "rp-production-nsg"
+	// deploy managed identity and set id
+	rpServicePrincipalID, err = d.deployManageIdentity(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return rpServicePrincipalID, d.deployPreDeploy(ctx, rpServicePrincipalID)
+}
+
+func (d *deployer) deployPreDeploy(ctx context.Context, rpServicePrincipalID string) (err error) {
+	deploymentName := "rp-production-predeploy"
+	_, err = d.deployments.Get(ctx, d.resourceGroup, deploymentName)
+	if isDeploymentNotFoundError(err) {
```

Comment:
`if !isDeploymentNotFoundError(err) { return err }` saves you an indent --> better; you can then also remove the `var b []byte` below and do `b, err := Asset(...)`